### PR TITLE
Refactor plugin installer

### PR DESCRIPTION
**Pull request by helenkzhang**

### CODE REVIEW

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Checksum/SHA256DirectoryChecksumCalculator.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /// <summary>
+    ///     Represents a checksum calculator that uses SHA256.
+    /// </summary>
+    public sealed class SHA256DirectoryChecksumCalculator
+        : IDirectoryChecksumCalculator
+    {
+        /// <inheritdoc />
+        public async Task<string> GetDirectoryChecksumAsync(string directory)
+        {
+            return await HashUtils.GetDirectoryHashAsync(directory);
```
Comment (mslukebo) [inline src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Checksum/SHA256DirectoryChecksumCalculator.cs:18]: You might want to move the impl here so it's obvious that it MUST be a SHA256 (given the class name)

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/PluginsSystem.cs [hunk 2]
```diff
@@ -84,13 +92,15 @@ public sealed class PluginsSystem
         /// </returns>
         public static PluginsSystem CreateFileBasedPluginsSystem(
             string root,
+            Func<string, Task<bool>> loadPluginFromDirectory,
```
Comment (mslukebo) [inline src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/PluginsSystem.cs:95]: we might want to consider, in the future, just making this an interface `IPluginDirectoryLoader`

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/IPluginsStorageDirectory.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public interface IPluginsStorageDirectory
+    {
+        /// <summary>
+        ///     Gets the dedicated directory where the given plugin is installed or should be installed.
+        /// </summary>
+        /// <param name="pluginIdentity">
+        ///     The identity of the plugin.
+        /// </param>
+        /// <returns>
+        ///     The directory where the given plugin is installed or should be installed.
```
Comment (mslukebo) [inline src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/IPluginsStorageDirectory.cs:22]: I wonder if we want these methods to be returning _relative_ paths. That way the implementation doesn't need to prepend some "root directory," but rather the consumers do. Thoughts?

Comment (helenkzhang) [reply src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/IPluginsStorageDirectory.cs:22]: Returning relative paths does make it more flexible but this interface has a lot of consumers (storage, loader, remover, checksum calculator and validator) and all of them will need to know about the plugins system root and append relative paths if we make this change. And these components are supposed to be sharing one root directory. Would it be better to just have `IPluginsStorageDirectory` to store the root?

Comment (mslukebo) [reply src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/IPluginsStorageDirectory.cs:22]: That's a good point. I'd say leave it as-is and have this class store the root

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Checksum/SHA256DirectoryChecksumCalculator.cs (hunk 1 above)
Comment (helenkzhang) [reply src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Checksum/SHA256DirectoryChecksumCalculator.cs]: Updated

**Review by mslukebo — APPROVED** [review]

**Pull request merged**